File: src/main/java/tk/skulk/plugin/core/extensions/messageoverride/MessageOverrideExtension.java
package tk.skulk.plugin.core.extensions.messageoverride;

import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.Plugin;
import tk.skulk.plugin.core.extensions.messageoverride.listeners.PlayerAdvancementDoneListener;
import tk.skulk.plugin.core.extensions.messageoverride.listeners.PlayerDeathListener;
import tk.skulk.plugin.core.extensions.messageoverride.listeners.PlayerJoinQuitListener;
import tk.skulk.plugin.core.extensions.messageoverride.listeners.UnknownCommandListener;
import tk.skulk.plugin.wrapper.BaseExtension;

public final class MessageOverrideExtension extends BaseExtension {
    public MessageOverrideExtension(final @NotNull Plugin plugin) {
        super(plugin);
    }

    @Override protected void initListeners() {
        new PlayerAdvancementDoneListener(this);
        new PlayerDeathListener(this);
        new PlayerJoinQuitListener(this);
        new UnknownCommandListener(this);
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/messageOverride/MessageOverrideExtension.kt
package tk.skulk.plugin.core.extensions.messageOverride

import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.messageOverride.listeners.PlayerDeathListener
import tk.skulk.plugin.core.extensions.messageOverride.listeners.PlayerJoinQuitListener
import tk.skulk.plugin.core.extensions.messageOverride.listeners.UnknownCommandListener
import tk.skulk.plugin.framework.Extension

class MessageOverrideExtension(plugin: SNPlugin) : Extension<SNPlugin>(plugin) {
    override val listeners = listOf(
        { PlayerDeathListener(this) },
        { PlayerJoinQuitListener(this) },
        { UnknownCommandListener(this) },
    )
}
